src/TraceEvent/NativeDlls.cs

```diff
@@ -15,7 +15,9 @@ class NativeDlls
     /// <param name="simpleName"></param>
     public static void LoadNative(string simpleName)
     {
-        var thisDllDir = Path.GetDirectoryName(Assembly.GetExecutingAssembly().ManifestModule.FullyQualifiedName);
+        // When TraceEvent is loaded as embedded assembly the executing assembly is mscorlib which points to the wrong path.
```

Comment:
I have several tools which rely on TraceEvent in different versions. I would like to embed TraceEvent and related things and load them as byte arrays into the AppDomain so I do not need to deploy the referenced assemblies to prevent breaking any tool relying on TraceEvent. 

I just have found that OSExtensions.dll
`private static void LoadKernelTraceControl()
{
	if (!ETWKernelControl.s_KernelTraceControlLoaded)
	{
		try
		{
			string arg_27_0 = Path.GetDirectoryName(Assembly.GetExecutingAssembly().ManifestModule.FullyQualifiedName);
`

uses the same trick which breaks the approach. I know that these wrapper are searched relative to the dll but if it is loaded from a byte array the Manifest path will be <Unknown> which then breaks the Path.Combine call due to invalid characters. 
A non breaking workaround would be to check first the returned ManifestModule.FullyQualifiedName for <Unkown>. If this is bad then use the executable directory as fallback in TraceEvent and OsExtensions. That way the old deployments are not broken and new ones are enabled.